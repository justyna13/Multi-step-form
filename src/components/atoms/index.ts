export { FormStep } from "@/components/atoms/FormStep/FormStep.tsx";
export { Button } from "@/components/atoms/Button/Button.tsx";
export { Input } from "@/components/atoms/Input/Input.tsx";
export { Header } from "@/components/atoms/Header/Header.tsx";
