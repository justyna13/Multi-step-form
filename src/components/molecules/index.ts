export { FormNavigation } from "@/components/molecules/FormNavigation/FormNavigation.tsx";
export { Footer } from "@/components/molecules/Footer/Footer.tsx";
export { FormInput } from "@/components/molecules/FormInput/FormInput.tsx";
export { FormSummary } from "@/components/molecules/FormSummary/FormSummary.tsx";
