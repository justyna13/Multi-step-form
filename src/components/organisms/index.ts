export { FormStepFirst } from '@/components/organisms/FormStepFirst/FormStepFirst.tsx';
export { FormStepSecond } from '@/components/organisms/FormStepSecond/FormStepSecond.tsx';
export { FormStepThird } from '@/components/organisms/FormStepThird/FormStepThird.tsx';
export { FormStepFourth } from '@/components/organisms/FormStepFourth/FormStepFourth.tsx';
export { FormConfirmation } from '@/components/organisms/FormConfirmation/FormConfirmation.tsx';
