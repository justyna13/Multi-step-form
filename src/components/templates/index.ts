export { HomePageTemplate } from '@/components/templates/HomePageTemplate/HomePageTemplate.tsx';
