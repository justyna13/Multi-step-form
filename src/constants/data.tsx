import { AVAILABLE_PLANS } from "@/constants/index.ts";
import { SinglePlanType } from "@/components/organisms/FormStepSecond/FormStepSecond.tsx";
import { ReactComponent as ArcadePlanIcon } from "@/assets/icons/icon-arcade.svg";
import { ReactComponent as AdvancedPlanIcon } from "@/assets/icons/icon-advanced.svg";
import { ReactComponent as ProPlanIcon } from "@/assets/icons/icon-pro.svg";
import { SingleAddOn } from "@/components/organisms/FormStepThird/FormStepThird.tsx";

export const availablePlans: Array<SinglePlanType> = [
  {
    id: AVAILABLE_PLANS.ARCADE,
    name: "Arcade",
    pricePerMonth: 9,
    icon: <ArcadePlanIcon />,
  },
  {
    id: AVAILABLE_PLANS.ADVANCED,
    name: "Advanced",
    pricePerMonth: 12,
    icon: <AdvancedPlanIcon />,
  },
  {
    id: AVAILABLE_PLANS.PRO,
    name: "Pro",
    pricePerMonth: 15,
    icon: <ProPlanIcon />,
  }
];

export const availableAddOns: Array<SingleAddOn> = [
  {
    id: 1,
    name: "Online service",
    description: "Access to multiplayer games",
    pricePerMonth: 1,
    fieldName: "onlineService"
  },
  {
    id: 2,
    name: "Larger storage",
    description: "Extra 1TB of cloud save",
    pricePerMonth: 2,
    fieldName: "largeStorage"
  },
  {
    id: 3,
    name: "Customizable Profile",
    description: "Custom theme on your profile",
    pricePerMonth: 2,
    fieldName: "customizableProfile"
  }
];
