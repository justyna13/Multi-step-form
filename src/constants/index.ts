const withBasePath = (path: string) =>
  `${import.meta.env.VITE_APP_BASE_URL || ''}${path}`;

export const Slugs = {
  // Main pages
  FORM: withBasePath('/multiStepForm'),
  NOT_FOUND: '*'
};

export enum FORM_STEPS {
  INFO = 'INFO',
  PLAN = 'PLAN',
  ADDITIONAL = 'ADDITIONAL ADD-ONS',
  SUMMARY = 'SUMMARY'
}

export enum AVAILABLE_PLANS {
  ARCADE = 'Arcade',
  ADVANCED = 'Advanced',
  PRO = 'Pro'
}

export enum PAYMENT_PERIODS {
  PER_MONTH = 'PER_MONTH',
  PER_YEAR = 'PER_YEAR'
}

export enum AVAILABLE_ADDONS {
  ONLINE_SERVICE = 'onlineService',
  LARGE_STORAGE = 'largeStorage',
  CUSTOMIZABLE_PROFILE = 'customizableProfile'
}
