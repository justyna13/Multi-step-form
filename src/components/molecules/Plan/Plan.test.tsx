import { render } from '@testing-library/react';
import { describe, expect } from 'vitest';

import { ReactComponent as ArcadePlanIcon } from '@/assets/icons/icon-arcade.svg';
import { Plan } from '@/components/molecules';
import { AVAILABLE_PLANS } from '@/constants';

const examplePlan = {
  id: AVAILABLE_PLANS.ARCADE,
  name: 'Arcade',
  pricePerMonth: 9,
  icon: <ArcadePlanIcon />
};

describe('Plan', () => {
  it('should match snapshot', () => {
    const view = render(
      <Plan
        testid="testid"
        item={examplePlan}
        isSelected={false}
        onPlanSelected={() => {}}
      />
    );

    expect(view).toMatchSnapshot();
  });
});
