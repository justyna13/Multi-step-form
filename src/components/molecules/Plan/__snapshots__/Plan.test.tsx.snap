// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Plan > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="single-plan "
        data-testid="testid"
      >
        <svg
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <circle
              cx="20"
              cy="20"
              fill="#FFAF7E"
              r="20"
            />
            <path
              d="M24.995 18.005h-3.998v5.998h-2v-5.998H15a1 1 0 0 0-1 1V29a1 1 0 0 0 1 1h9.995a1 1 0 0 0 1-1v-9.995a1 1 0 0 0-1-1Zm-5.997 8.996h-2v-1.999h2v2Zm2-11.175a2.999 2.999 0 1 0-2 0v2.18h2v-2.18Z"
              fill="#FFF"
              fill-rule="nonzero"
            />
          </g>
        </svg>
        <div>
          <p
            class="single-plan__ttl"
          >
            Arcade
          </p>
          <p
            class="single-plan__desc"
          >
            $
            9
            /mo
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="single-plan "
      data-testid="testid"
    >
      <svg
        height="40"
        viewBox="0 0 40 40"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <circle
            cx="20"
            cy="20"
            fill="#FFAF7E"
            r="20"
          />
          <path
            d="M24.995 18.005h-3.998v5.998h-2v-5.998H15a1 1 0 0 0-1 1V29a1 1 0 0 0 1 1h9.995a1 1 0 0 0 1-1v-9.995a1 1 0 0 0-1-1Zm-5.997 8.996h-2v-1.999h2v2Zm2-11.175a2.999 2.999 0 1 0-2 0v2.18h2v-2.18Z"
            fill="#FFF"
            fill-rule="nonzero"
          />
        </g>
      </svg>
      <div>
        <p
          class="single-plan__ttl"
        >
          Arcade
        </p>
        <p
          class="single-plan__desc"
        >
          $
          9
          /mo
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
